internal/hcp/internal/controllers/link/controller.go

```diff
@@ -15,17 +19,69 @@ import (
 	pbhcp "github.com/hashicorp/consul/proto-public/pbhcp/v2"
 )
 
-func LinkController(resourceApisEnabled bool, hcpAllowV2ResourceApis bool) *controller.Controller {
+type HCPClientFn func(link *pbhcp.Link) (hcpclient.Client, error)
```

Comment:
When should this type be used? Can we add a comment.

Reply:
Done! Let me know if the comment is clear.

```
// HCPClientFn is a function that can be used to create an HCP client from a Link object.
// This function type should be passed to a LinkController in order to tell it how to make a client from
// a Link. For normal use, DefaultHCPClientFn should be used, but tests can substitute in a function that creates a
// mock client.
```